Sources/Sentry/SentryMetricProfiler.mm

```diff
@@ -113,7 +113,7 @@ @implementation SentryMetricProfiler {
 - (instancetype)initWithMode:(SentryProfilerMode)mode
 {
     if (self = [super init]) {
-        [self clear];
+        [self clearNotThreadSafe];
```

Comment:
l: Why we need a notThreadSafe function in here?

Reply:
It doesn't make sense to have a synchronized block in the initializer. That's why.